### DISCUSSION TITLE
How to read cookie on server side without getServerSideProps and api routes?

### DISCUSSION THREAD

**Discussion by arso1er** [vercel/next.js#19607] [Help] [answered]
Hi,

I use nextjs with a custom express server and I'd like to read a cookie value on the server.  The following code used in the server setup makes me think nextjs has access to the request object from express:
 `
nextApp.prepare().then(() => {
  app.get('*', (req, res) => {
    return handle(req, res);
  });
});
`
How can I have access to the request object, or at least the cookies. I don't need to set cookies, just read them. 

Thanks.

**Comment by jamesmosier** [comment, marked as answer]
Hi @arso1er. If you are using an express server then you can use the `cookie-parser` package for express: http://expressjs.com/en/resources/middleware/cookie-parser.html

```js
const cookieParser = require('cookie-parser');

nextApp.prepare().then(() => { 
  const app = express();

  app.use(cookieParser());

  app.get('*', (req, res) => { 
    const cookies = req.cookies;
    return handle(req, res); 
  }); 
});
```

Or I believe without the `cookie-parser` you can access the cookies via `req.headers.cookie` as a string.

**Comment by arso1er** [comment]
Hi @jamesmosier. Yes I'm already using `cookie-parser` and working with cookies in my express api. What I wanted to know is if there's a way to read the cookies in a nextjs page without using getServerSideProps (I'm using getStaticProps). But now that I think more about it, static pages cached on a cdn can't have access to cookies.

Thanks for your answer.
